pkg/prometheus/resource_selector.go

```diff
@@ -691,6 +691,15 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			continue
 		}
 
+		for _, rl := range sc.Spec.MetricRelabelConfigs {
```

Comment:
Option 1 isn't good since we'd skip the rejection function if the `action` field is empty.

I haven't looked deeply into the code, but option 3 could indeed be a change in functionality for other ServiceMonitors, PodMonitors and Probes. Let's keep that in a separate PR, if relevant 😅 

Option 2 looks like a good suggestion, but I can see that we're already using this same pattern every other place we call `validateRelabelConfig`. It would be nice to change those in a separate PR if you feel like it :)

Reply:
Yeah!
It's totally fine for me keep as it is and we do the changes we think it's valuable in another PR 💪🏽